src/lightning_app/core/flow.py

```diff
@@ -634,3 +636,10 @@ def my_remote_method(self, name):
             lightning my_command_name --args name=my_own_name
         """
         raise NotImplementedError
+
+    def on_save_state_dict(self) -> Optional[Dict[str, Any]]:
+        """Override to return any required metadata for reloading."""
+        return None
+
+    def on_load_state_dict(self, children_states: Dict[str, Any], extras: Optional[Dict[str, Any]] = None) -> None:
+        """Override to add your re-instantiation logic for dynamic components."""
```

Comment:
I think we can find better names for these methods. I can't think of a name aTM :D

Reply:
I think the docstring should explicitly say that here is the place to re-instantiate dynamic components. 
And it should also show how you are supposed to call super